python/ray/_private/monitor.py

```diff
@@ -95,8 +96,11 @@ def __init__(self,
         self.redis = ray._private.services.create_redis_client(
             redis_address, password=redis_password)
 
+        (ip, port) = redis_address.split(":")
+        self.gcs_client = connect_to_gcs(ip, int(port), redis_password)
```

Comment:
I think we'll have gcs running here, right? In L105, it's gcs_channel. If we have gcs running, it should work.

Reply:
But it will be the 1.2.0 GCS server, so how will it know about the `InternalKVGcsService`?

Reply:
What's the plan here for autoscaler compatibility with old clusters? The only solution that comes to mind is to preserve the existing Redis-based internal KV and fall back to that in the autoscaler if a test ping through the internal KV accessor fails.

Reply:
I think we can just revert the changes to this file entirely, then once we cut off support for 1.2.0, do this + flip the flag to use in-memory store together.

Reply:
> But it will be the 1.2.0 GCS server, so how will it know about the InternalKVGcsService?
Do you mean autoscaler and ray will share the same gcs? Then I understood it incorrectly. 

This will only work if autoscaler uses (autoscaler doesn't upgrade) the ray GCS.